### PR TITLE
pangram: Slim down descriptions to generate better test identifiers 

### PR DESCRIPTION
**Pull request by bencoman**
The purpose of the "description" field is to derive identifiers for naming generated test methods, but some descriptions have tended to verbosity ending up with long identifiers (up to 150 characters). Such long method names are harder for students to work with and also are awkward for some IDE GUIs.

Issue #1473 proposed adding a separate "identifier" field, but I was required to first try improving "descriptions" to better suit "identifier" generation. This PR aims to slim down the generated identifiers without losing substantive information about the test.

To start with,  "Check if the given string is a pangram" is a rather banal redundant description unworthy of nesting.

### CODE REVIEW

**Comment by bencoman** [comment]
I had split this into two commits to make it easier to separate review nesting and description changes.
Just discovered you get a nice overall view if you go...
   Files changed > Settings (gear icon) > Hide whitespace changes

File: exercises/pangram/canonical-data.json [hunk 1]
```diff
@@ -1,97 +1,90 @@
...
+    {
+      "description": "only lower case",
+      "property": "isPangram",
+      "input": {
+        "sentence": "the quick brown fox jumps over the lazy dog"
+      },
+      "expected": true
+    },
+    {
+      "description": "missing character 'x'",
```
Comment (rpottsoh) [inline exercises/pangram/canonical-data.json]:
```suggestion
      "description": "missing an 'x'",
```
or perhaps
```suggestion
      "description": "lacks an 'x'",
```

Similar pattern could be repeated if others are on board.  Also, the whole file essentially appears to have been altered.  Some may find it more difficult to review.

**Comment by bencoman** [comment]
btw, The strategy I use to create slim descriptions is imagine the word "Test" preceding the description (because including it would be redundant)

**Comment by rpottsoh** [comment]
> Files changed > Settings (gear icon) > Hide whitespace changes

Nice find.  TIL something, and it is only the morning.

File: exercises/pangram/canonical-data.json (hunk 1 above)
Comment (bencoman) [reply exercises/pangram/canonical-data.json]: I think "lacks" might be harder for non-native-english speakers to grok.

**Comment by bencoman** [comment]
@rpottsoh I went to push a similar change to your suggestion into `missing character 'h'  `
but realised that camelCase identifier convention would generate `missing a 'h' `  into `missingAH` 
which is not nice.  I reverted your suggestion so it will generate `missingCharacterH` .
Let me know if you object.

File: exercises/pangram/canonical-data.json (hunk 1 above)
Comment (kotp) [reply exercises/pangram/canonical-data.json]: Perhaps `"Missing the letter 'x'"`

Comment (rpottsoh) [reply exercises/pangram/canonical-data.json]: @bencoman `missingTheLetterX` or `missingAnX`?  I agree `lacksAnX` is the least desirable choice.

Comment (bencoman) [reply exercises/pangram/canonical-data.json]: In general I think scaffold words like "the" and "an" can often be left out without compromising meaning - but I can slide on it.  The majority of improvement from this PR comes in removing the nesting platitude description "Check if the given string is a pangram" and shrinking the description "upper and lower case versions of the same character should not be counted separately".  Then once I opened the box, other descriptions get some collateral cleaning (like removing redundant 'panagram' from descriptions.

As for length, my gut feel was 40-60 but I thought it would be good to put some science behind it by reviewing the stock Pharo 7 release...
```smalltalk
classes := Object allSubclasses select: [ :cc | cc isKindOf: TestCase class ].
methods := c flatCollect: [ :c | c allMethods  ].
tests := methods select: [ :m | m selector beginsWith: 'test' ].
lengths := tests collect: [ :m | m selector size ].
lengths asBag keysAndValuesDo: [ :len :count | Transcript crShow: len; show: ','; show: count  ]
```
and graphing in Excel...
image

So 40-60 looks like a reasonable target to aim for, but its not an absolute. I arbitrarily picked fixing descriptions longer than 100 chars.  When I get there I'll be able to better evaluate if more is needed. (and it looks like I should review of Pharo's longer ones also)

**Comment by bencoman** [comment]
@kotp, @rpottsoh, I committed your suggestion. Anything further be giving this the tick?

File: exercises/pangram/canonical-data.json [hunk 2]
```diff
@@ -1,97 +1,90 @@
...
-      ]
+      "description": "empty sentence",
+      "property": "isPangram",
+      "input": {
+        "sentence": ""
+      },
+      "expected": false
+    },
+    {
+      "description": "perfect lower case",
```
Comment (ErikSchierboom) [inline exercises/pangram/canonical-data.json]: I was wondering if we should maybe rephrase this test case or add something to the README to describe what a perfect pangram is.

Comment (kotp) [reply exercises/pangram/canonical-data.json]: That sounds like a good idea. (different PR though for the README.  `"perfect pangram in lower case"` might be a better description of the test

Comment (kotp) [reply exercises/pangram/canonical-data.json]: Context being the pangram, which is why I think that the wording ended up being the way it became.

Comment (SleeplessByte) [reply exercises/pangram/canonical-data.json]: I am in favour of adding to the readme as wel.

Comment (ErikSchierboom) [reply exercises/pangram/canonical-data.json]: @kotp @SleeplessByte If we do a separate PR for the readme, is this PR ok to be merged?

Comment (kotp) [reply exercises/pangram/canonical-data.json]: Yes, I think so.

**Comment by SleeplessByte** [comment]
Another one merged @bencoman 🚀

**Comment by bencoman** [comment]
thx all.
